quesma/processors/es_to_ch_ingest/elasticsearch_to_clickhouse_ingest_processor.go

```diff
@@ -92,70 +105,90 @@ func (p *ElasticsearchToClickHouseIngestProcessor) prepareTemporaryIngestProcess
 	tableDisco := clickhouse.NewTableDiscovery(oldQuesmaConfig, connectionPool, virtualTableStorage)
 	schemaRegistry := schema.NewSchemaRegistry(clickhouse.TableDiscoveryTableProviderAdapter{TableDiscovery: tableDisco}, oldQuesmaConfig, clickhouse.SchemaTypeAdapter{})
 
-	ip := ingest.NewIngestProcessor2(oldQuesmaConfig, chBackendConn, nil, tableDisco, schemaRegistry, virtualTableStorage, esBackendConn)
+	dummyTableResolver := table_resolver.NewDummyTableResolver()
+
+	ip := ingest.NewIngestProcessor(oldQuesmaConfig, connectionPool, nil, tableDisco, schemaRegistry, virtualTableStorage, dummyTableResolver)
 
 	ip.Start()
 	return ip
 }
 
 func (p *ElasticsearchToClickHouseIngestProcessor) Handle(metadata map[string]interface{}, message ...any) (map[string]interface{}, any, error) {
 	var data []byte
-	indexNameFromIncomingReq := metadata[IngestTargetKey].(string)
-	if indexNameFromIncomingReq == "" {
-		fmt.Printf("Missing index name in metadata") // SHOULD NEVER HAPPEN AND NOT BE VERIFIED HERE I GUESS
-		return nil, data, nil
-	}
 
 	for _, m := range message {
-		mCasted, err := quesma_api.CheckedCast[*quesma_api.Request](m)
+		req, err := quesma_api.CheckedCast[*http.Request](m)
 		if err != nil {
 			fmt.Printf("ElasticsearchToClickHouseIngestProcessor: invalid message type: %v", err)
 			return nil, data, err
 		}
-		messageAsHttpReq := mCasted.OriginalRequest
 
-		if _, present := p.config.IndexConfig[indexNameFromIncomingReq]; !present && metadata[IngestAction] == DocIndexAction {
+		var pathPattern string
+		if val, ok := metadata[es_to_ch_common.PathPattern]; ok {
+			pathPattern = val.(string)
+		}
+		indexPatterFromRequestUri := es_to_ch_common.GetParamFromRequestURI(req, pathPattern, "index")
+
+		if _, present := p.config.IndexConfig[indexPatterFromRequestUri]; !present && pathPattern == es_to_ch_common.IndexDocPath {
 			// `_doc` at this point can go directly to Elasticsearch,
 			// `_bulk` request might be still sent to ClickHouse as the req payload may contain documents targeting CH tables
-			resp := p.legacyIngestProcessor.SendToElasticsearch(messageAsHttpReq)
-			respBody, err := ReadResponseBody(resp)
-			if err != nil {
-				println(err)
-			}
-			return metadata, respBody, nil
+			return p.routeToElasticsearch(metadata, req)
 		}
 
-		bodyAsBytes, err := frontend_connectors.ReadRequestBody(messageAsHttpReq)
-		if err != nil {
+		var reqBodyBytes []byte
+		if reqBodyBytes, err = frontend_connectors.ReadRequestBody(req); err != nil {
 			panic("ElasticsearchToClickHouseIngestProcessor: invalid message type")
 		}
 
-		switch metadata[IngestAction] {
-		case DocIndexAction:
-			payloadJson, err := types.ExpectJSON(types.ParseRequestBody(string(bodyAsBytes)))
+		// TODO this comes from Quesma general config and should be passed to the processor
+		ingestStats := false
```

Comment:
Do you plan to to be on par with old functionality in this PR?

Reply:
I am not able to w.r.t. v2 ingest processor (at least at this point). Enabling ingest stats is a processor property which unfortunately - happens to live as a general quesma config setting - so we'd have to address this somehow at the config level. But that's IMO out of scope of this PR